logging/jul/src/main/java/io/helidon/logging/jul/JulMdc.java

```diff
@@ -15,22 +15,24 @@
  */
 package io.helidon.logging.jul;
 
-import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Supplier;
 
 /**
  * MDC implementation for Java Util Logging framework.
  */
 public class JulMdc {
 
-    private static final ThreadLocal<Map<String, String>> MDC_PROPERTIES = ThreadLocal.withInitial(HashMap::new);
-
     private JulMdc() {
         throw new IllegalStateException("This class cannot be instantiated");
     }
 
     static void put(String key, String value) {
-        MDC_PROPERTIES.get().put(key, value);
+        JulMdcProvider.instance().put(key, value);
```

Comment:
Do not use a singleton in any of the implementations. All methods are static and should be used as such.

Reply:
No longer an issue with the changes to the PR.